WardrobeCloset/Controllers/AccessoriesController.cs

```diff
@@ -0,0 +1,132 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Data.Entity;
+using System.Linq;
+using System.Net;
+using System.Web;
+using System.Web.Mvc;
+using WardrobeCloset.Models;
+
+namespace WardrobeCloset.Controllers
+{
+    public class AccessoriesController : Controller
+    {
+        private WardrobeClosetProjectEntities db = new WardrobeClosetProjectEntities();
+
+        // GET: Accessories
+        public ActionResult Index()
+        {
+            var accessories = db.Accessories.Include(a => a.Outfit);
+            return View(accessories.ToList());
+        }
+       
+        // GET: Accessories/Details/5
+        public ActionResult Details(int? id)
+        {
+            if (id == null)
+            {
+                return new HttpStatusCodeResult(HttpStatusCode.BadRequest);
```

Comment:
Nice! Having a missing id would be a client error (of which bad request is one). I presume this controller would be called if the user accessed a path like "/Accessories/Details".

What might be valuable to add would be the actual problem that occurred. From [the API description](https://msdn.microsoft.com/en-us/library/system.web.mvc.httpstatuscoderesult.httpstatuscoderesult(v=vs.118).aspx) of the `HttpStatusCodeResult`, it appears that you could do something like:
```
return new HttpStatusCodeResult(HttpStatusCode.BadRequest, "Your request was missing an accessory id".
```